src/node_crypto.cc

```diff
@@ -1134,6 +1175,46 @@ void SecureContext::LoadPKCS12(const FunctionCallbackInfo<Value>& args) {
 }
 
 
+#ifndef OPENSSL_NO_ENGINE
+void SecureContext::SetClientCertEngine(
+    const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+  CHECK_EQ(args.Length(), 1);
+  CHECK(args[0]->IsString());
+
+  SecureContext* sc = Unwrap<SecureContext>(args.This());
+
+  MarkPopErrorOnReturn mark_pop_error_on_return;
+
+  // SSL_CTX_set_client_cert_engine does not itself 'support' multiple
+  // calls by cleaning up before overwriting the client_cert_engine
+  // internal context variable.
+  // Instead of trying to fix up this problem we in turn also do not
+  // support multiple calls to SetClientCertEngine. As this may surprise
+  // a developer we throw him an exception.
+  if (sc->ctx_->client_cert_engine != nullptr) {
```

Comment:
This is arguably better done through a flag on the `SecureContext` rather than groping around inside openssl's internals.  I guess it's okay for now although it might become an issue when we upgrade openssl.

Reply:
Will this perhaps already cause a problem if Node.js is compiled with shared lib OpenSSL?

Reply:
It might, yes.

Reply:
Probably worth fixing right now, then, I imagine...